pointgrey_camera_driver/src/nodelet.cpp

```diff
@@ -158,34 +160,38 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
     // Check if we should disconnect (there are 0 subscribers to our data)
     if(it_pub_.getNumSubscribers() == 0 && pub_->getPublisher().getNumSubscribers() == 0)
     {
-      NODELET_DEBUG("Disconnecting.");
-      pubThread_->interrupt();
-      scopedLock.unlock();
-      pubThread_->join();
-      scopedLock.lock();
-      sub_.shutdown();
-
-      try
-      {
-        NODELET_DEBUG("Stopping camera capture.");
-        pg_.stop();
-      }
-      catch(std::runtime_error& e)
+      if (pubThread_)
```

Comment:
All the previous code is now indented inside this `if` and the only other change is `pubThread_ = NULL;`